config/authorization/config.ex

```diff
@@ -21,16 +21,45 @@ defmodule Acl.UserGroups.Config do
       query: sparql_query_for_access_role( group_string ) }
   end
 
-  defp sparql_query_for_access_role( group_string ) do
-    "PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
-    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
-    SELECT DISTINCT ?session_group ?session_role WHERE {
-      <SESSION_ID> ext:sessionGroup/mu:uuid ?session_group;
-                   ext:sessionRole ?session_role.
-      FILTER( ?session_role = \"#{group_string}\" )
-    }"
+  defp sparql_query_for_access_role(group_string) do
+    """
+      PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+      PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+      PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+      PREFIX muAccount: <http://mu.semte.ch/vocabularies/account/impersonation/>
+
+      SELECT DISTINCT ?session_group ?session_role WHERE {
+
+        VALUES ?session_role {
+           \"#{group_string}\"
+        }
+
+        VALUES ?session_id {
+           <SESSION_ID>
+        }
+
+       {
+          ?session_id ext:sessionGroup/mu:uuid ?own_group;
+                        ext:sessionRole ?session_role.
+       } UNION {
+
+         ?session_id muAccount:impersonates ?maybe_impersonated.
+
+         ?maybe_impersonated a foaf:OnlineAccount;
+           ext:sessionRole ?session_role;
+           foaf:accountServiceHomepage <https://github.com/lblod/mock-login-service>.
+
+         ?person a foaf:Person;
+           foaf:account ?maybe_impersonated;
+           foaf:member/mu:uuid ?maybe_impersonated_group.
+       }
+       BIND(COALESCE(?maybe_impersonated_group, ?own_group) AS ?session_group)
+      }
+      LIMIT 1
```

Comment:
Small style nitpick:
```suggestion
      PREFIX ext:       <http://mu.semte.ch/vocabularies/ext/>
      PREFIX mu:        <http://mu.semte.ch/vocabularies/core/>
      PREFIX foaf:      <http://xmlns.com/foaf/0.1/>
      PREFIX muAccount: <http://mu.semte.ch/vocabularies/account/impersonation/>

      SELECT DISTINCT ?session_group ?session_role WHERE {
        VALUES ?session_role {
          \"#{group_string}\"
        }

        VALUES ?session_id {
          <SESSION_ID>
        }

        {
          ?session_id ext:sessionGroup/mu:uuid ?own_group;
            ext:sessionRole ?session_role.
        }
        UNION
        {
          ?session_id muAccount:impersonates ?maybe_impersonated.
          ?maybe_impersonated a foaf:OnlineAccount;
            ext:sessionRole ?session_role;
            foaf:accountServiceHomepage <https://github.com/lblod/mock-login-service>.

          ?person a foaf:Person;
            foaf:account ?maybe_impersonated;
            foaf:member/mu:uuid ?maybe_impersonated_group.
        }

        BIND(COALESCE(?maybe_impersonated_group, ?own_group) AS ?session_group)
      } LIMIT 1
```